### ISSUE TITLE
When I retrain the model using the parameters of best_trial,  I cannot achieve the F1 of optuna turning result

### ISSUE THREAD

**Issue by chaiyixuan** [optuna/optuna#3247]
### Environment

- Optuna version:
- Python version:3.8
- OS:linux
- (Optional) Other libraries and their versions:

### Error messages, stack traces, or logs

```shell
When I retrain the model using the parameters of best_trial,  I cannot achieve the F1 of optuna turning result. The best_trial model F1 is 84%, but i use the parameters of best_trial to retrain a new model, the F1 is 80%
```

### Steps to reproduce

1. When I retrain the model using the parameters of best_trial,  I cannot achieve the F1 of optuna turning result
2.
3.
```python
# python code
```

**Comment by chaiyixuan** [comment]
And I have set the random seed by pytorch_lighting.seed_everything()

**Comment by nzw0301** [comment]
Hi, could you share minimal reproducible codes with us?

**Comment by chaiyixuan** [comment]
`
if version.parse(pl.__version__) < version.parse("1.0.2"):
    raise RuntimeError("PyTorch Lightning>=1.0.2 is required for this example.")

PERCENT_VALID_EXAMPLES = 0.1
BATCHSIZE = 128
CLASSES = 10
EPOCHS = 10
DIR = os.getcwd()


class Net(nn.Module):
    def __init__(self, dropout: float, output_dims: List[int]):
        super().__init__()
        layers: List[nn.Module] = []

        input_dim: int = 28 * 28
        for output_dim in output_dims:
            layers.append(nn.Linear(input_dim, output_dim))
            layers.append(nn.ReLU())
            layers.append(nn.Dropout(dropout))
            input_dim = output_dim

        layers.append(nn.Linear(input_dim, CLASSES))

        self.layers: nn.Module = nn.Sequential(*layers)

    def forward(self, data: torch.Tensor) -> torch.Tensor:
        logits = self.layers(data)
        return F.log_softmax(logits, dim=1)


class LightningNet(pl.LightningModule):
    def __init__(self, dropout: float, output_dims: List[int]):
        super().__init__()
        self.model = Net(dropout, output_dims)

    def forward(self, data: torch.Tensor) -> torch.Tensor:
        return self.model(data.view(-1, 28 * 28))

    def training_step(self, batch, batch_idx: int) -> torch.Tensor:
        data, target = batch
        output = self(data)
        return F.nll_loss(output, target)

    def validation_step(self, batch, batch_idx: int) -> None:
        data, target = batch
        output = self(data)
        pred = output.argmax(dim=1, keepdim=True)
        accuracy = pred.eq(target.view_as(pred)).float().mean()
        self.log("val_acc", accuracy)
        print(accuracy)
        self.log("hp_metric", accuracy, on_step=False, on_epoch=True)
        return accuracy
    
    def validation_epoch_end(self, outputs) -> None:
        acc_list = []
        for accuracy in outputs:
          acc_list.append(accuracy)
        score = sum(acc_list)/len(acc_list)
        self.log("epoch_val", score)
        print("epoch_val",score) 

    def configure_optimizers(self) -> optim.Optimizer:
        return optim.Adam(self.model.parameters())


class FashionMNISTDataModule(pl.LightningDataModule):
    def __init__(self, data_dir: str, batch_size: int):
        super().__init__()
        self.data_dir = data_dir
        self.batch_size = batch_size

    def setup(self, stage: Optional[str] = None) -> None:
        self.mnist_test = datasets.FashionMNIST(
            self.data_dir, train=False, download=True, transform=transforms.ToTensor()
        )
        mnist_full = datasets.FashionMNIST(
            self.data_dir, train=True, download=True, transform=transforms.ToTensor()
        )
        self.mnist_train, self.mnist_val = random_split(mnist_full, [55000, 5000])

    def train_dataloader(self) -> DataLoader:
        return DataLoader(
            self.mnist_train, batch_size=self.batch_size, shuffle=True, pin_memory=True
        )

    def val_dataloader(self) -> DataLoader:
        return DataLoader(
            self.mnist_val, batch_size=self.batch_size, shuffle=False, pin_memory=True
        )

    def test_dataloader(self) -> DataLoader:
        return DataLoader(
            self.mnist_test, batch_size=self.batch_size, shuffle=False, pin_memory=True
        )

`

**Comment by nzw0301** [comment]
I think you need to call `seed_everything` in `objective`.
